http/sys_health.go

```diff
@@ -114,13 +114,28 @@ func getSysHealth(core *vault.Core, r *http.Request) (int, *HealthResponse, erro
 		code = standbyCode
 	}
 
+	// Fetch the local cluster name and identifier
+	var clusterName, clusterID string
+	cluster, err := core.Cluster()
+
+	// Don't set the cluster details in the health status when Vault is sealed
+	if err != nil && err.Error() != "Vault is sealed" {
```

Comment:
I think a better approach would be to use the `sealed` boolean from earlier, and simply skip the cluster checks if it's true. That way if we know that the core isn't sealed but get an error anyways, we can just assume it's an internal error (and not rely on string matching).